iphone/Maps/Core/Location/MWMLocationManager.mm

```diff
@@ -469,43 +492,27 @@ - (void)setStarted:(BOOL)started
   }
 }
 
-- (BOOL)start
+- (void)requestPermissionIfNeededFor:(CLLocationManager *)manager
 {
-  MWMVoidBlock doStart = ^{
-    LOG(LINFO, ("startUpdatingLocation"));
-
-    CLLocationManager * locationManager = self.locationManager;
-    if ([locationManager respondsToSelector:@selector(requestWhenInUseAuthorization)])
-      [locationManager requestWhenInUseAuthorization];
-
-    [locationManager startUpdatingLocation];
-
-    setPermissionRequested();
+  [manager requestWhenInUseAuthorization];
+  setPermissionRequested();
+}
 
-    if ([CLLocationManager headingAvailable])
-      [locationManager startUpdatingHeading];
-  };
+- (void)startUpdatingLocationFor:(CLLocationManager *)manager
+{
+  [self requestPermissionIfNeededFor:manager];
+  
+  LOG(LINFO, ("startUpdatingLocation"));
+  
+  [manager startUpdatingLocation];
```

Comment:
I'm still not understanding why `requestPermissionIfNeededFor` is needed here. If there is no permission granted, and `requestPermissionIfNeededFor` is called, it won't immediately grant the permission anyway (it will be done async), so the following call to `[manager startUpdatingLocation];` should fail, right?

Is there a way to _know_ the current permission status and avoid calling "if needed" requests?

https://developer.apple.com/documentation/corelocation/requesting_authorization_to_use_location_services/#4061636